gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/purger/HivePurgerQueryTemplate.java

```diff
@@ -0,0 +1,193 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.compliance.purger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+
+import gobblin.compliance.utils.PartitionUtils;
+
+
+/**
+ * This class creates all queries required by {@link HivePurgerConverter}
+ *
+ * @author adsharma
+ */
+public class HivePurgerQueryTemplate {
+
+  /**
+   * Use the staging db for creating staging tables.
+   * Alter query doesn't work with dbName.tableName.
+   */
+  public static String getUseDbQuery(String dbName) {
+    return "USE " + dbName;
+  }
+
+  public static String getAutoConvertJoinProperty() {
+    return "SET hive.auto.convert.join=false";
+  }
+
+  /**
+   * Will allow insert query to specify only partition coloumn names instead of partition spec.
+   */
+  public static String getDynamicPartitionNonStrictModeProperty() {
+    return "SET hive.exec.dynamic.partition.mode=nonstrict";
+  }
+
+  /**
+   * To get around the OOM, we need to set hive.optimize.sort.dynamic.partition to true so that the data is sorted by
+   * partition key and at one time only one partition needs to be accessed.
+   */
+  public static String getOptimizeSortDynamicPartition() {
+    return "SET hive.optimize.sort.dynamic.partition=true";
+  }
+
+  /**
+   * If staging table doesn't exist, it will create a staging table.
+   */
+  public static String getCreateTableQuery(String completeNewTableName, String likeTableDbName, String likeTableName,
+      String location) {
+    return getCreateTableQuery(completeNewTableName, likeTableDbName, likeTableName) + " LOCATION " + PartitionUtils
+        .getQuotedString(location);
+  }
+
+  public static String getCreateTableQuery(String completeNewTableName, String likeTableDbName, String likeTableName) {
+    return "CREATE EXTERNAL TABLE IF NOT EXISTS " + completeNewTableName + " LIKE " + likeTableDbName + "."
+        + likeTableName;
+  }
+
+  /**
+   * This query will create a partition in staging table and insert the datasets whose compliance id is not
+   * contained in the compliance id table.
+   */
+  public static String getInsertQuery(PurgeableHivePartitionDataset dataset) {
+    return "INSERT OVERWRITE" + " TABLE " + dataset.getCompleteStagingTableName() + " PARTITION (" + PartitionUtils
+        .getPartitionSpecString(dataset.getSpec()) + ")" + " SELECT /*+MAPJOIN(b) */ " + getCommaSeperatedColumnNames(
```

Comment:
NOT IN is not equivalent to left join/NULL. Handling of null fields is different.